org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/engine/connected/ConnectedEngineFacade.java

```diff
@@ -364,6 +367,17 @@ public void updateProjectList(IProgressMonitor monitor) {
     });
   }
 
+  @Override
+  public Set<String> getBoundProjectKeys() {
+    return ProjectsProviderUtils.allProjects().stream()
```

Comment:
Maybe you can share part of this code with the following method `getBoundProjects`?